storage/Filesystems/BtrfsSubvolumeImpl.cc

```diff
@@ -193,14 +193,14 @@ namespace storage
     MountOpts
     BtrfsSubvolume::Impl::get_default_mount_options() const
     {
-	return MountOpts(vector<string>({ "subvol=/" + path }));
+	return MountOpts(vector<string>({ "subvol=" + quote("/" + path) }));
```

Comment:
Duh. You are right. Silly me.

Reply:
Oops - no; it's even worse.

Since this is about mount options, you cannot simply use single or double quotes; the fstab format has a very specific way of quoting; see _man fstab_. The EtcFstab class already handles this. If you add quotes here, this will result in a syntax error.

So this **needs** to remain without quotes.

Reply:
https://github.com/openSUSE/libstorage-ng/blob/master/storage/EtcFstab.cc#L242

Reply:
It turns out that it's unspecified how this should be handled; _man fstab_ only mentions the device and the mount point field and how to use `\040` for a space there. I could not find yet any specification for those subvolume names.

Reply:
_man 5 btrfs_ also only says

> subvol=path
>
> Mount subvolume from path rather than the toplevel subvolume. The path is always treated as relative to the toplevel subvolume. This mount option overrides the default subvolume set for the given filesystem.

No mention how to handle special characters.

But if we allow a blank here, even if it is inside single or double quotes, it is clearly a syntax error for the fstab file format.

Reply:
For the time being, please use `EtcFstab::fstab_encode()` instead of this `quote()`. This is the only safe way that I can see.

If there are no blanks in the path, there is no harm done. If there are, at least it won't cause a syntax error for parsing the entire fstab and thus a complete DoS for booting that system.

Reply:
Please notice that it will still fail if somebody uses a comma which is the separator for mount options. I don't think fstab can handle that at all.

Reply:
I was playing with it, and I see that it works with command `mount` but not as fstab entry:

* This works: `mount subvol='/@/tmp' /tmp`
* But this fstbab entry does not work: `UUID=121213 /tmp btrfs subvol='/@/tmp'`

So, as you suggested, I will use `EtcFstab::fstab_encode()`.